src/ucm/mmap/install.c

```diff
@@ -102,46 +102,60 @@ void ucm_fire_mmap_events(int events)
     }
 }
 
-/* Called with lock held */
-static ucs_status_t ucm_mmap_test(int events)
+static ucs_status_t ucm_mmap_test_events(int events, int *out_events)
 {
-    static int installed_events = 0;
     ucm_event_handler_t handler;
-    int out_events;
-
-    if (ucs_test_all_flags(installed_events, events)) {
-        /* All requested events are already installed */
-        return UCS_OK;
-    }
 
     /* Install a temporary event handler which will add the supported event
      * type to out_events bitmap.
      */
     handler.events   = events;
     handler.priority = -1;
     handler.cb       = ucm_mmap_event_test_callback;
-    handler.arg      = &out_events;
-    out_events       = 0;
+    handler.arg      = out_events;
+    *out_events      = 0;
 
     ucm_event_handler_add(&handler);
 
     ucm_fire_mmap_events(events);
 
     ucm_event_handler_remove(&handler);
 
-    /* TODO check address / stop all threads */
-    installed_events |= out_events;
-    ucm_debug("mmap test: got 0x%x out of 0x%x, total: 0x%x", out_events, events,
-              installed_events);
+    ucm_debug("mmap test: got 0x%x out of 0x%x", *out_events, events);
 
     /* Return success iff we caught all wanted events */
-    if (!ucs_test_all_flags(out_events, events)) {
+    if (!ucs_test_all_flags(*out_events, events)) {
         return UCS_ERR_UNSUPPORTED;
     }
 
     return UCS_OK;
 }
 
+ucs_status_t ucm_test_events(int events)
+{
+    int out_events;
+
+    return ucm_mmap_test_events(events, &out_events);
```

Comment:
ucm_mmap_test -> `/* Called with lock held */`
ucm_mmap_test_events is called from ucm_mmap_test -> also need such comment ?
this function is API, should it take lock?

also, should't we test malloc events as well?

Reply:
> ucm_mmap_test_events is called from ucm_mmap_test -> also need such comment ?

no, it should work out of lock - we are reproducing actual case which may be caused by user

there are no malloc events